File: public/public-web/src/main/java/ch/difty/scipamato/publ/web/authentication/LoginPage.java
package ch.difty.scipamato.publ.web.authentication;

import com.giffing.wicket.spring.boot.context.scan.WicketSignInPage;
import org.apache.wicket.request.mapper.parameter.PageParameters;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.wicketstuff.annotation.mount.MountPath;

import ch.difty.scipamato.common.web.pages.login.AbstractLoginPage;
import ch.difty.scipamato.publ.web.paper.browse.PublicPage;

@MountPath("login")
@WicketSignInPage
public class LoginPage extends AbstractLoginPage<PublicPage> {

    private static final long serialVersionUID = 1L;

    public LoginPage(@Nullable PageParameters parameters) {
        super(parameters);
    }

    @NotNull
    @Override
    protected PublicPage getResponsePage() {
        return new PublicPage(getPageParameters());
    }
}

File: public/public-web/src/main/kotlin/ch/difty/scipamato/publ/web/authentication/LoginPage.kt
package ch.difty.scipamato.publ.web.authentication

import ch.difty.scipamato.common.web.pages.login.AbstractLoginPage
import ch.difty.scipamato.publ.web.paper.browse.PublicPage
import com.giffing.wicket.spring.boot.context.scan.WicketSignInPage
import org.apache.wicket.request.mapper.parameter.PageParameters
import org.wicketstuff.annotation.mount.MountPath

@MountPath("login")
@WicketSignInPage
class LoginPage(
    parameters: PageParameters?,
) : AbstractLoginPage<PublicPage>(parameters) {

    override val responsePage: PublicPage
        get() = PublicPage(pageParameters)

    companion object {
        private const val serialVersionUID = 1L
    }
}
